test/jdk/java/awt/print/PrinterJob/ImagePrinting/ImageTypes.java

```diff
@@ -21,85 +21,151 @@
  * questions.
  */
 
-/**
- *
+/*
  * @test
  * @bug 4521945 7006865
  * @summary Test printing images of different types.
  * @author prr
- * @run main/manual=yesno/timeout=900 ImageTypes
+ * @run main/manual ImageTypes
  */
 
-import java.io.*;
-import static java.awt.Color.*;
-import java.awt.*;
-import java.awt.geom.*;
-import java.awt.event.*;
-import java.awt.print.*;
-import java.awt.image.*;
-import static java.awt.image.BufferedImage.*;
-import javax.print.*;
-import javax.print.attribute.*;
-import javax.print.attribute.standard.*;
-
-public class ImageTypes extends Frame implements ActionListener {
-
-    private ImageCanvas c;
-
-    public static void main(String args[]) {
-
-        ImageTypes f = new ImageTypes();
-        f.show();
+import java.awt.Button;
+import java.awt.Color;
+import java.awt.Component;
+import java.awt.Dimension;
+import java.awt.Font;
+import java.awt.FontMetrics;
+import java.awt.Frame;
+import java.awt.GradientPaint;
+import java.awt.Graphics;
+import java.awt.Graphics2D;
+import java.awt.GridLayout;
+import java.awt.Panel;
+import java.awt.TextArea;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.image.BufferedImage;
+import java.awt.image.DataBuffer;
+import java.awt.image.IndexColorModel;
+import java.awt.print.PageFormat;
+import java.awt.print.Printable;
+import java.awt.print.PrinterException;
+import java.awt.print.PrinterJob;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import javax.print.attribute.HashPrintRequestAttributeSet;
+import javax.print.attribute.PrintRequestAttributeSet;
+
+import static java.awt.Color.black;
+import static java.awt.Color.blue;
+import static java.awt.Color.cyan;
+import static java.awt.Color.darkGray;
+import static java.awt.Color.gray;
+import static java.awt.Color.green;
+import static java.awt.Color.lightGray;
+import static java.awt.Color.magenta;
+import static java.awt.Color.orange;
+import static java.awt.Color.pink;
+import static java.awt.Color.red;
+import static java.awt.Color.white;
+import static java.awt.Color.yellow;
+import static java.awt.image.BufferedImage.TYPE_3BYTE_BGR;
+import static java.awt.image.BufferedImage.TYPE_4BYTE_ABGR;
+import static java.awt.image.BufferedImage.TYPE_4BYTE_ABGR_PRE;
+import static java.awt.image.BufferedImage.TYPE_BYTE_BINARY;
+import static java.awt.image.BufferedImage.TYPE_BYTE_GRAY;
+import static java.awt.image.BufferedImage.TYPE_BYTE_INDEXED;
+import static java.awt.image.BufferedImage.TYPE_INT_ARGB;
+import static java.awt.image.BufferedImage.TYPE_INT_ARGB_PRE;
+import static java.awt.image.BufferedImage.TYPE_INT_BGR;
+import static java.awt.image.BufferedImage.TYPE_INT_RGB;
+import static java.awt.image.BufferedImage.TYPE_USHORT_555_RGB;
+import static java.awt.image.BufferedImage.TYPE_USHORT_GRAY;
```

Comment:
Thats lot of static imports from just two classes. This just looks weird. Is it recommended practice to do? I am not saying this is wrong as you are removing wild card imports, but it just does not look good to have so many statement for importing from two classes.

Reply:
In most of the test cases wild card is removed and its replaced with each class import statement.

Reply:
Found the java doc about the static import. In this case we can use a wild card instead of importing all the static imports. 
https://docs.oracle.com/javase/7/docs/technotes/guides/language/static-import.html